### ISSUE TITLE
vscode.env.openExternal is not working correctly because of unnecessary escape processing.

### ISSUE THREAD

**Issue by wraith13** [microsoft/vscode#76606]
- VSCode Version: v1.36 and 1.36.0-insider
- OS Version: Windows, Mac

Steps to Reproduce:

Just excute this code in a extention.

```typescript
vscode.env.openExternal(vscode.Uri.parse("x-github-client://openRepo/https://github.com/wraith13/open-in-github-desktop-vscode.git"));
```

`x-github-client` is a [GitHub Desktop](https://desktop.github.com/)'s protocol.
GitHub Desktop receives `openRepo/https%3A//github.com/wraith13/open-in-github-desktop-vscode.git` and fail. ( Must be `openRepo/https://github.com/wraith13/open-in-github-desktop-vscode.git` )

Does this issue occur when all extensions are disabled?: No ( It is API for extensions )

It worked fine with v1.35.
I think that it may be a problem of electron.

**Comment by jrieken** [comment]
This is a regression from https://github.com/microsoft/vscode/commit/781e61bf568a5e99756597c21ae2e3d3d293b1a0 (which itself fixes a regression from https://github.com/microsoft/vscode/pull/73849). Next try...

**Comment by eamodio** [comment]
@jrieken I think this needs to be re-opened. With the latest code, there doesn't seem to be a way for me to open a url that contains a `#` in it (that isn't a fragment).

For example in GitLens, when if you try to open a branch named something like `fixes-#44396` on a remote, I can't get it to work anymore with `openExternal` because of the double escaping. I escape the `fixes-%2344396` so that it becomes `https://github.com/eamodio/vscode/commits/fixes-%2344396`, then I `Uri.parse` it which all looks good:
```
g {scheme: "https", authority: "github.com", path: "/eamodio/vscode/commits/fixes-#44396", query: "", fragment: "", …}
_formatted:null
_fsPath:null
authority:"github.com"
fragment:""
fsPath:"\eamodio\vscode\commits\fixes-#44396"
path:"/eamodio/vscode/commits/fixes-#44396"
query:""
scheme:"https"
```

But then I pass that Uri to `openExternal` on it encodes the url again, so it tries to open `https://github.com/eamodio/vscode/commits/fixes-%252344396` which doesn't exist.

I don't see any way around this.

**Comment by jrieken** [comment]
> @jrieken I think this needs to be re-opened.

It is (re-opened) https://github.com/microsoft/vscode/issues/25852
